### PR TITLE
Allow compilation in AVR only mode

### PR DESCRIPTION
**Pull request by arunh**
This patch allows us to take advantage of the Arduino boot loader and of course CMake without allowing the code any dependency on Arduino libraries.

This option is very useful if you are using an Arduino board as a way to speed up development of programs designed to be as small as possible and compliant with the Atmel chip range outside of the Arduino environment.

### CODE REVIEW

**Comment by queezythegreat** [comment]
Hi, this nicely integrates the functionaly I was planing in #44.

**Comment by arunh** [comment]
Missed #44 but looks good. I don't think it will be too hard to amend my pull to fit what is described there.

**Comment by queezythegreat** [comment]
Could you update your changes to fit the ticket (the option name)? I'll close that ticket and merge in your changes.

File: example/CMakeLists.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ generate_arduino_example(Wire master_writer)
 # Original blink sketch (from Arduino SDK examples)
 #====================================================================#
 generate_arduino_firmware(blink_original
-    SKETCH ${ARDUINO_SDK_PATH}/examples/1.Basics/Blink
+    SKETCH ${ARDUINO_SDK_PATH}/examples/01.Basics/Blink
```
Comment (arunh) [inline example/CMakeLists.txt:21]: I can't be sure this fixes for all version/platforms. But latest Mac Arduino version requires this name.

**Comment by arunh** [comment]
Updates made, and also code added to allow 'MANUAL' mode in library creation. Tested OK on my code base.

**Comment by queezythegreat** [comment]
I've just merged in your changes into the develop branch. The blink example did not work for me, but I added work around.

**Comment by arunh** [comment]
Thanks!

On 14 Oct 2012, at 15:28, Tomasz Bogdal notifications@github.com wrote:

I've just merged in your changes into the develop branch. The blink example
did not work for me, but I added work around.

—
Reply to this email directly or view it on
GitHubhttps://github.com/queezythegreat/arduino-cmake/pull/50#issuecomment-9419950.

**Pull request closed without merging**